Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -106,13 +107,21 @@ public ActionsTreeView(VisualElement root, StateContainer stateContainer)
                 treeViewItem.EditTextFinished -= treeViewItem.EditTextFinishedCallback;
             };
 
-            m_ActionsTreeView.selectedIndicesChanged += indicies =>
+            m_ActionsTreeView.selectedIndicesChanged += indices =>
             {
-                var index = indicies.First();
-                if (index == -1)
-                    return;
-                var item = m_ActionsTreeView.GetItemDataForIndex<ActionOrBindingData>(index);
-                Dispatch(item.isAction ? Commands.SelectAction(item.name) : Commands.SelectBinding(item.bindingIndex));
+                if (!m_DeselectionHelper.Select(m_ActionsTreeView, indices))
```

Comment:
Nitpick: Would it make more sense to have this as something like `m_SelectionHelper.Deselect(...)` , `m_SelectionHelper.Disable(..)` or `m_SelectionHelper.DisableSelection(..)`? Just because it feels we're using a lot "select" wording all around so it would be nice if we could be consistent. Having "deselection" kind of seems counterintuitive.
But maybe it's really needed to have this "deselection" wording, I didn't do a full deep dive on this so I might lack context anyway.

Reply:
I do not like the name DeselectionHelper either but failed to come up with a better name at the time of writing. The reason I named it Select is rather that it may be transparently returning a selection index or acting upon the current index or deselecting. Will try to come up with a better name..... would e.g. `m_SelectionFilter.Apply(...)` be better since it actually filters the selection change and modifies it if needed ? Typically for other UI frameworks the behaviour I am preventing here is typically called "de-selection"